speedtest/request.go

```diff
@@ -136,79 +148,134 @@ func (s *Server) UploadTest(savingMode bool) error {
 	return nil
 }
 
-func dlWarmUp(dlURL string) error {
+func dlWarmUp(ctx context.Context, dlURL string) error {
 	size := dlSizes[2]
 	xdlURL := dlURL + "/random" + strconv.Itoa(size) + "x" + strconv.Itoa(size) + ".jpg"
 
-	resp, err := client.Get(xdlURL)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, xdlURL, nil)
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+
 	if err != nil {
 		return err
 	}
+
 	defer resp.Body.Close()
-	ioutil.ReadAll(resp.Body)
 
-	return nil
+	_, err = io.Copy(ioutil.Discard, resp.Body)
```

Comment:
Note: I changed this code slightly to prevent the whole response body being read into memory - instead it's read in chunks so the most memory that will be used will be (currently - this may change in future Go versions...) 32K. The whole response body will still be read, just with lower memory usage.

Reply:
Thank you for trying to reduce memory usage! Though on my testing, there is no memory impact 😢 
Before (master branch) : 129MB
After (this PR branch) : 125MB
I measure memory usage by `$ ps aux | grep speedtest`.

If you have some testing results, could you provide it to me? Even if you don't have, this change doesn't seems to have any bad effect, so I'll accept it 👍